### PR TITLE
Enable Cloud Build. Improve Dockerfile

### PR DESCRIPTION
**Pull request by huyhg**

### CODE REVIEW

File: cloudbuild-tag.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+- name: gcr.io/cloud-builders/docker
+  entrypoint: bash
+  args:
+  - -exc
+  - |
+    docker pull gcr.io/$_STAGING_PROJECT_ID/testrunner:sha_$COMMIT_SHA
+    docker tag  gcr.io/$_STAGING_PROJECT_ID/testrunner:sha_$COMMIT_SHA gcr.io/$PROJECT_ID/testrunner:$TAG_NAME
+
+images:
+- gcr.io/$P}ROJECT_ID/testrunner:$TAG_NAME
```
Comment (wgrzelak) [inline cloudbuild-tag.yaml]:
```suggestion
- gcr.io/$PROJECT_ID/testrunner:$TAG_NAME
```

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+# Copy source code and build
+
+COPY . /usr/local/src/testrunner
+WORKDIR /usr/local/src/testrunner
+
+# Validate OSS checksums
+
+RUN set -xeu && \
+    mkdir -p /usr/share/testrunner && \
+    wget -q -O /usr/share/testrunner/testrunner.LICENSE https://raw.githubusercontent.com/GoogleCloudPlatform/marketplace-k8s-app-tools/master/LICENSE && \
```
Comment (wgrzelak) [inline Dockerfile]:
```suggestion
    wget -q -O /usr/share/testrunner/testrunner.LICENSE https://raw.githubusercontent.com/GoogleCloudPlatform/marketplace-testrunner/master/LICENSE && \
```

File: cloudbuild-tag.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
+steps:
+
+- name: gcr.io/cloud-builders/docker
+  entrypoint: bash
+  args:
+  - -exc
+  - |
+    docker pull gcr.io/$_STAGING_PROJECT_ID/testrunner:sha_$COMMIT_SHA
+    docker tag  gcr.io/$_STAGING_PROJECT_ID/testrunner:sha_$COMMIT_SHA gcr.io/$PROJECT_ID/testrunner:$TAG_NAME
```
Comment (wgrzelak) [inline cloudbuild-tag.yaml]:
```suggestion
    docker tag gcr.io/$_STAGING_PROJECT_ID/testrunner:sha_$COMMIT_SHA gcr.io/$PROJECT_ID/testrunner:$TAG_NAME
```

File: cloudbuild-tag.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
+steps:
+
+- name: gcr.io/cloud-builders/docker
```
Comment (wgrzelak) [inline cloudbuild-tag.yaml]: Optionally - I would suggest to split this for two steps. Similar to https://github.com/GoogleCloudPlatform/marketplace-vm-imagebuilder/blob/master/cloudbuild.yaml#L37-L49.

File: cloudbuild.yaml [hunk 5]
```diff
@@ -11,36 +11,27 @@
...
 # See the License for the specific language governing permissions and
 # limitations under the License.
+timeout: 300s
+options:
+  machineType: 'N1_HIGHCPU_8'
 
 steps:
 
-# Run tests
+# Build all
```
Comment (wgrzelak) [inline cloudbuild.yaml]: Seems like this step is no necessary. The binary is built in "Build an image" step.

Comment (huyhg) [reply cloudbuild.yaml]: The build image step only builds specific targets. This is meant as a test that everything still compiles. I'll add a comment.

File: cloudbuild-tag.yaml (hunk 4 above)
Comment (huyhg) [reply cloudbuild-tag.yaml]: Done.

**Pull request merged**